consensus/scheme/rolldpos/roundcalculator.go

```diff
@@ -133,12 +133,11 @@ func (c *roundCalculator) roundInfo(
 ) (roundNum uint32, roundStartTime time.Time, err error) {
 	lastBlockTime := time.Unix(c.chain.GenesisTimestamp(), 0)
 	if height > 1 {
-		var lastBlock *block.Footer
-		if lastBlock, err = c.chain.BlockFooterByHeight(height - 1); err != nil {
+		var lastBlock *block.Header
+		if lastBlock, err = c.chain.BlockHeaderByHeight(height - 1); err != nil {
 			return
 		}
-		lastBlockCommitTime := lastBlock.CommitTime()
-		lastBlockTime = lastBlockTime.Add(lastBlockCommitTime.Sub(lastBlockTime) / c.blockInterval * c.blockInterval)
+		lastBlockTime = lastBlockTime.Add(lastBlock.Timestamp().Sub(lastBlockTime) / c.blockInterval * c.blockInterval)
```

Comment:
not sure whether we should do this. It may cause problem that delegates upgrade their nodes at different time, that they will have different "last block time", some use "commit time", some use "block time".
Potential solution: Only use block time after berling, and then delete it in the next version after berling.
Open to discussion.

Reply:
works for me, will have a PR to follow this check-in